packages/apps/asana/src/services/oauth.ts

```diff
@@ -1,113 +1,139 @@
+      description:
+        "Asana is a web and mobile application designed to help teams organize, track, and manage their work. It's a popular project management tool that enables teams to collaborate more effectively.",
       oauth_url: `https://app.asana.com/-/oauth_authorize?response_type=code&client_id=${client_id}&redirect_uri=${redirect}`,
-      slug:AppNameDefinitions.ASANA,
-      category:AppCategoryDefinitions.PRODUCTIVITY,
-      developer:"Ocular AI",
-      images:["/asana.svg"],
+      slug: AppNameDefinitions.ASANA,
+      category: AppCategoryDefinitions.PRODUCTIVITY,
+      developer: "Ocular AI",
+      images: ["/asana.svg"],
+      auth_strategy: auth_strategy,
       overview: "Asana is a web and mobile application designed to help teams.",
       docs: "https://developers.asana.com",
-      website: "https://asana.com"
-    }
+      website: "https://asana.com",
+    };
   }
 
   async refreshToken(refresh_token: string): Promise<OAuthToken> {
     const body = {
-      grant_type: 'refresh_token',
+      grant_type: "refresh_token",
       client_id: this.client_id_,
       client_secret: this.client_secret_,
       refresh_token: refresh_token,
-    }
+    };
 
     const config = {
       headers: {
         "content-type": "application/x-www-form-urlencoded",
       },
     };
 
-   return axios
-    .post("https://app.asana.com/-/oauth_token", 
-      Object.keys(body)
-    .map((key) => `${key}=${encodeURIComponent(body[key])}`)
-    .join('&'), 
-      config)
-    .then((res) => {
-      return {
-        token : res.data.access_token,
-        token_expires_at : new Date(Date.now() + res.data.expires_in * 1000),
-        refresh_token: res.data.refresh_token,
-        refresh_token_expires_at: new Date(Date.now() + 172800 * 1000),
-      } as OAuthToken
-    })
-    .catch((err) => {
-      console.error(err);
-      throw err;
-    });
+    return axios
+      .post(
+        "https://app.asana.com/-/oauth_token",
+        Object.keys(body)
+          .map((key) => `${key}=${encodeURIComponent(body[key])}`)
+          .join("&"),
+        config
+      )
+      .then((res) => {
+        return {
+          token: res.data.access_token,
+          token_expires_at: new Date(Date.now() + res.data.expires_in * 1000),
+          refresh_token: res.data.refresh_token,
+          refresh_token_expires_at: new Date(Date.now() + 172800 * 1000),
+        } as OAuthToken;
+      })
+      .catch((err) => {
+        console.error(err);
+        throw err;
+      });
   }
 
   async generateToken(code: string): Promise<OAuthToken> {
     console.log("***** Generating token from the code:\n");
-    
+
+    if (this.auth_strategy_ === AppAuthStrategy.API_TOKEN_STRATEGY) {
+      return {
+        type: "API token",
```

Comment:
Can you create an enum for the type here?

type: Enum.token

Reply:
ok , will create one in types!

Reply:
@louismurerwa